lib/topological_inventory/sync/inventory_upload/payload.rb

```diff
@@ -193,6 +193,48 @@ def sources_api_client(tenant = nil)
 
           SourcesApiClient::DefaultApi.new(api_client)
         end
+
+        class << self
+          def topological_api_client(tenant = nil)
+            @topological_api_client ||=
+              begin
+                api_client = TopologicalInventoryApiClient::ApiClient.new
+
+                if tenant
+                  x_rh_identity = Base64.strict_encode64(JSON.dump('identity' => { 'account_number' => tenant }))
+                  api_client.default_headers.merge!('x-rh-identity' => x_rh_identity)
+                end
+
+                TopologicalInventoryApiClient::DefaultApi.new(api_client)
```

Comment:
Total side-note but the newer openapi generated clients let you override the headers per api-call so we might be able to get away from this sort of stuff in the future and just use the normal DefaultApi